app/src/main/java/com/example/bisit/ui/myPageOwner/DialogAddCoupon.kt

```
@@ -0,0 +1,195 @@
+    private fun showDatePicker() {
+        val calendar = Calendar.getInstance()
+        val year = calendar.get(Calendar.YEAR)
+        val month = calendar.get(Calendar.MONTH)
+        val day = calendar.get(Calendar.DAY_OF_MONTH)
+
+        // Using standard DatePickerDialog with a cleaner theme and ensuring correct behavior
+        val datePickerDialog = DatePickerDialog(
+            context,
+            R.style.DatePickerTheme,
+            { _, selectedYear, selectedMonth, selectedDay ->
+                val dateStr = String.format("%04d.%02d.%02d", selectedYear, selectedMonth + 1, selectedDay)
```

detekt (1.23.8)

[warning] String.format("%04d.%02d.%02d", selectedYear, selectedMonth + 1, selectedDay) uses implicitly default locale for string formatting.

(detekt.potential-bugs.ImplicitDefaultLocale)